src/library/FOSSBilling/Environment.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+/**
+ * Copyright 2022-2023 FOSSBilling
+ * Copyright 2011-2021 BoxBilling, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * @copyright FOSSBilling (https://www.fossbilling.org)
+ * @license http://www.apache.org/licenses/LICENSE-2.0 Apache-2.0
+ */
+
+namespace FOSSBilling;
+
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\Dotenv\Exception\PathException;
+
+class Environment
+{
+    const ENV_KEY = 'APP_ENV';
+    const PRODUCTION = 'prod';
+    const DEVELOPMENT = 'dev';
+    const TESTING = 'test';
+
+    const POSSIBLE = [self::PRODUCTION, self::DEVELOPMENT, self::TESTING];
+    const DEFAULT = self::PRODUCTION;
+
+    /**
+     * Get the current environment of the application.
+     * The environment variable set in the operating system will have priority over the environment variable set in the .env file.
+     * 
+     * @return string
+     */
+    public function getCurrentEnvironment(): string
+    {
+        return in_array(getenv(self::ENV_KEY), self::POSSIBLE) ? getenv(self::ENV_KEY) : self::DEFAULT;
+    }
+
+    /**
+     * Check if the current environment is a production environment.
+     * @return bool
+     */
+    public function isProduction(): bool
+    {
+        return $this->getCurrentEnvironment() === self::PRODUCTION;
+    }
+
+    /**
+     * Check if the current environment is a development environment.
+     * @return bool
+     */
+    public function isDevelopment(): bool
+    {
+        return $this->getCurrentEnvironment() === self::DEVELOPMENT;
+    }
+
+    /**
+     * Check if the current environment is a testing environment.
+     * @return bool
+     */
+    public function isTesting(): bool
+    {
+        return $this->getCurrentEnvironment() === self::TESTING;
+    }
+
+    /**
+     * Check if the current environment is a CLI environment.
+     * @return bool
+     */
+    public function isCLI(): bool
+    {
+        return php_sapi_name() === 'cli';
```

Comment:
I think this should actually be:
```PHP
return (php_sapi_name() === 'cli' || !http_response_code()) ? true : false;
```

From the [http_response_code docs](https://www.php.net/manual/en/function.http-response-code.php):
> false will be returned if response_code is not provided and it is not invoked in a web server environment (such as from a CLI application).

I only state this because I don't know if `php_sapi_name` is 100% the end-all be-all for checking if PHP is running from CLI. See this, for example: https://stackoverflow.com/questions/10886539/why-does-php-sapi-not-equal-cli-when-called-from-a-cron-job